admin/aws.py

```diff
@@ -0,0 +1,249 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Effectful interface to boto.
+"""
+
+from characteristic import attributes, Attribute
+from effect import sync_performer, TypeDispatcher
+
+import boto
+
+
+@attributes([
+    "bucket",
+    "prefix",
+    "target_prefix",
+])
+class UpdateS3RoutingRule(object):
+    """
+    Update a routing rule for an S3 bucket website endpoint to point to a new
+    path.
+
+    If the path is changed, return the old path.
+
+    :ivar bytes bucket: Name of bucket to change routing rule for.
+    :ivar bytes prefix: Prefix to change routing rule for.
+    :ivar bytes target_prefix: Target prefix to redirect to.
+    """
+
+
+@sync_performer
+def perform_update_s3_routing_rule(dispatcher, intent):
+    """
+    See :class:`UpdateS3RoutingRule`.
```

Comment:
Looking at that class's docstring (just above this) I don't see parameter documentation for either of these parameters.